protected/models/Patient.php

```diff
@@ -311,6 +311,30 @@ public function getPatientSource()
         return 'None';
     }
 
+    /**
+     * @param $status string Patient's status in a trial.
+     * @return string Number of trials the patient is in where they have the given status within a trial. This is returned as a string to preserve max. precision.
+     */
+    public function getTrialCount($status)
+    {
+        $criteria = new CDbCriteria();
+        $criteria->compare('patient_status', $status);
+        $criteria->compare('patient_id', $this->id);
+        return TrialPatient::model()->count($criteria);
+    }
+
+    /**
+     * @param $trial_id int Trial ID.
+     * @return string|null Patient's status in the given trial.
+     */
+    public function getTrialStatus($trial_id)
+    {
+        $criteria = new CDbCriteria();
+        $criteria->compare('trial_id', $trial_id);
+        $criteria->compare('patient_id', $this->id);
+        return TrialPatient::model()->find($criteria)->getStatusForDisplay();
```

Comment:
As mentioned in another review, this could be a useful function to have if the TrialPatient record was returned instead of their status.